librz/analysis/data.c

```diff
@@ -280,25 +251,40 @@ RZ_API void rz_analysis_data_free(RzAnalysisData *d) {
 	}
 }
 
-RZ_API RzAnalysisData *rz_analysis_data(RzAnalysis *analysis, ut64 addr, const ut8 *buf, int size, int wordsize) {
+/**
+ * @brief      { function_description }
+ *
+ * @param      analysis  The analysis
+ * @param[in]  addr      The address
+ * @param[in]  buf       The buffer
+ * @param[in]  size      The size
+ * @param[in]  wordsize  The wordsize
+ *
+ * @return     { description_of_the_return_value }
+ */
+RZ_API RZ_OWN RzAnalysisData *rz_analysis_data(RZ_NONNULL RzAnalysis *analysis, ut64 addr, RZ_NONNULL const ut8 *buf, int size, int wordsize) {
+	rz_return_val_if_fail(analysis && buf, NULL);
+
 	ut64 dst = 0;
-	int n, nsize = 0;
+	RzDetectedString *dstr = NULL;
+	RzStrEnc encoding = analysis->binb.bin ? rz_str_enc_string_as_type(analysis->binb.bin->strenc) : RZ_STRING_ENC_GUESS;
+	int n = 0;
 	int bits = analysis->bits;
-	int word = wordsize ? wordsize : RZ_MIN(8, bits / 8);
+	int word = wordsize > 0 ? wordsize : RZ_MIN(8, bits / 8);
 
 	if (size < 4) {
 		return NULL;
 	}
 	if (size >= word && is_invalid(buf, word)) {
-		return rz_analysis_data_new(addr, RZ_ANALYSIS_DATA_TYPE_INVALID, -1, buf, word);
+		return rz_analysis_data_new(addr, RZ_ANALYSIS_DATA_INFO_TYPE_INVALID, -1, buf, word);
 	}
 	{
-		int i, len = RZ_MIN(size, 64);
+		int len = RZ_MIN(size, 64);
```

Comment:
Isn't `size_t` a better fit here? Or `ut64` if we want to support really huge strings

Reply:
yes.